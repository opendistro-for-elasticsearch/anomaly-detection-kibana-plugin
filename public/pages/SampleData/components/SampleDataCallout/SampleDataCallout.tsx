/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { EuiCallOut, EuiLink } from '@elastic/eui';
import { APP_PATH, PLUGIN_NAME } from '../../../../utils/constants';

export const SampleDataCallout = () => {
  return (
    <EuiCallOut
      title="Looking to get more familiar with anomaly detection?"
      color="primary"
      iconType="help"
    >
      <p>
        Read the{' '}
        <EuiLink
          href="https://opendistro.github.io/for-elasticsearch-docs/docs/ad/"
          target="_blank"
        >
          documentation
        </EuiLink>{' '}
        or create a{' '}
        <EuiLink href={`${PLUGIN_NAME}#${APP_PATH.SAMPLE_DETECTORS}`}>
          sample detector
        </EuiLink>{' '}
        to get started.
      </p>
    </EuiCallOut>
  );
};
