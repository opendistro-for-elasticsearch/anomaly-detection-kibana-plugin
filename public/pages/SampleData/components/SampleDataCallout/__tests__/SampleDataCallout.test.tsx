/*
 * Copyright 2020 Amazon.com, Inc. or its /*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { SampleDataCallout } from '../SampleDataCallout';

describe('<SampleDataCallout /> spec', () => {
  describe('Data not loaded', () => {
    test('renders component', () => {
      const { container, getByText } = render(<SampleDataCallout />);
      expect(container.firstChild).toMatchSnapshot();
      getByText('Looking to get more familiar with anomaly detection?');
    });
  });
});
