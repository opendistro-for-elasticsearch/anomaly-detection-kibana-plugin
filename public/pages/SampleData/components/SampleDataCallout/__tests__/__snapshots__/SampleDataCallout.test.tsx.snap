// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleDataCallout /> spec Data not loaded renders component 1`] = `
<div
  class="euiCallOut euiCallOut--primary"
>
  <div
    class="euiCallOutHeader"
  >
    <svg
      aria-hidden="true"
      class="euiIcon euiIcon--medium euiIcon-isLoading euiCallOutHeader__icon"
      focusable="false"
      height="16"
      role="img"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    />
    <span
      class="euiCallOutHeader__title"
    >
      Looking to get more familiar with anomaly detection?
    </span>
  </div>
  <div
    class="euiText euiText--small"
  >
    <p>
      Read the
       
      <a
        class="euiLink euiLink--primary"
        href="https://opendistro.github.io/for-elasticsearch-docs/docs/ad/"
        rel="noopener noreferrer"
        target="_blank"
      >
        documentation
      </a>
       
      or create a
       
      <a
        class="euiLink euiLink--primary"
        href="opendistro-anomaly-detection-kibana#/sample-detectors"
        rel="noreferrer"
      >
        sample detector
      </a>
       
      to get started.
    </p>
  </div>
</div>
`;
